From 3160d2f2e9c4a6ba1f56907b35fe8bde96c6c751 Mon Sep 17 00:00:00 2001
From: dhineshkumar <mdhineshkumar@ami.com>
Date: Fri, 25 Aug 2023 18:46:00 +0530
Subject: [PATCH] Fixed virtual media incorrect UDC devices

Issue:
#1 - Media redirection using 1e6a0000.usb-vhub:p6 UDC not working
#2 - Path to check UDC used status changed after kernel 6.x

Fix:
#1 - Skipped 1e6a0000.usb-vhub:p6 UDC for media for now
#2 - Updated path to check used UDC status

Signed-off-by: Dhineshkumar M <mdhineshkumar@ami.com>
---
 src/system.hpp | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/src/system.hpp b/src/system.hpp
index 0ab1fa5..4180565 100644
--- a/src/system.hpp
+++ b/src/system.hpp
@@ -544,10 +544,27 @@ struct UsbGadget
                 for (const auto& port : fs::directory_iterator(
                          "/sys/bus/platform/devices/1e6a0000.usb-vhub"))
                 {
+                    std::string portId = port.path().filename();
+                    std::string gadgetSuspended = "gadget/suspended";
+                    if (!portId.compare("1e6a0000.usb-vhub:p1") ||
+                        !portId.compare("1e6a0000.usb-vhub:p2") ||
+                        !portId.compare("1e6a0000.usb-vhub:p3") ||
+                        !portId.compare("1e6a0000.usb-vhub:p4") ||
+                        !portId.compare("1e6a0000.usb-vhub:p5") ||
+                        !portId.compare("1e6a0000.usb-vhub:p7") )
+                    {
+                        std::string n = std::string(1, portId.back());
+                        int nn = std::stoi(n)-1;
+                        gadgetSuspended = "gadget." + std::to_string(nn) + "/suspended";
+                        LogMsg(Logger::Debug,"gadgetSuspended : ", gadgetSuspended);
+                    }
+                    else
+                    {
+                        continue;
+                    }
                     if (fs::is_directory(port) && !fs::is_symlink(port) &&
-                        !fs::exists(port.path() / "gadget/suspended"))
+                        !fs::exists(port.path() / gadgetSuspended))
                     {
-                        const std::string portId = port.path().filename();
                         LogMsg(Logger::Debug,
                                "Use port : ", port.path().filename());
                         echoToFile(gadgetDir / "UDC", portId);
-- 
2.34.1

